File: 2894_Divisible_And_Non-divisible_Sums_Difference.cpp
class Solution {
public:
    int differenceOfSums(int n, int m) {
        int S = n * (n + 1) / 2;
        int D = (n / m) * (n / m + 1) / 2 * m;
        return S - 2 * D;
    }
};

File: 2894_Divisible_And_Non-divisible_Sums_Difference.java
class Solution {
    public int differenceOfSums(int n, int m) {
        
        int r = 0, f = 0;
        for (int p = 1; p <= n; ++p) {
            if (p % m != 0) r += p;
            else f += p;
        }
        return r - f;
    }
}
